### PR TITLE
Add tests for effects

### PR DESCRIPTION
**Pull request by Pikachu920**
### Description
This is a PR to add tests for as many effects as I can.

---
**Target Minecraft Versions:** any
**Requirements:** N/A
**Related Issues:** #6158

### CODE REVIEW

File: build.gradle
Comment (Pikachu920) [inline build.gradle:43]: it seems like 5.2.0 has a bug. on v5.0.1, the action bar test passes. however, when running this exact same code in v5.2.0, the action bar test encounters the exception described in https://github.com/easymock/easymock/issues/356. that issue was apparently fixed in 5.2.0 but i wonder if it wasn't a complete fix. it could definitely be user error too though. i will probably followup on that issue and see what the maintainers think.

**Review by sovdeeth — COMMENTED** [review]
I don't think I know enough about EasyMock to comment on that code with authority, but everything looks alright from a glance. This seems like a very useful tool for player-related testing!
I may come back and actually approve once I know a bit more about how easymock works.

**Comment by Pikachu920** [comment]
> I don't think I know enough about EasyMock to comment on that code with authority, but everything looks alright from a glance. This seems like a very useful tool for player-related testing! I may come back and actually approve once I know a bit more about how easymock works.

it's really pretty simple! i don't know how it's actually implemented, but it's basically like a proxy that keeps track of it's method calls. You use `expect` to tell it what calls you are expecting, then you use `replay` to say "start listening for the expected calls", then you use `verify` to say "did the expected calls occur?". it can also do more complex things like ensuring the arguments match some criteria and stuff like that.

**Review by Moderocky — APPROVED** [review]
Nice work!

**Review by sovdeeth — APPROVED** [review]

File: build.gradle
Comment (Pikachu920) [reply build.gradle:43]: https://github.com/easymock/easymock/issues/356#issuecomment-1872626625

Comment (Pikachu920) [reply build.gradle:43]: [#515](https://github.com/easymock/easymock/issues/515)

**Comment by henri-tremblay** [comment]
Please try with EasyMock 5.4.0-SNAPSHOT. It shoud be fixed.

**Pull request merged**
